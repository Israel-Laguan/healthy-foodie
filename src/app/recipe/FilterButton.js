import React from "react";
import { Row, Col, ButtonGroup, Button } from "reactstrap";

function FilterButton({ items, onClick }) {
  return (
    <Row>
      <Col md={11}>
        <ButtonGroup className="w-100">
          {items.map((item) => (
            <Button
              key={`filter__button-${item.text}`}
              color="primary"
              active={item.actived}
              onClick={() => onClick(item.text)}
            >
              <img src={item.icon} alt={`Icon ${item.text}`} />
              {item.text}
            </Button>
          ))}
        </ButtonGroup>
      </Col>
    </Row>
  );
}

export default FilterButton;
