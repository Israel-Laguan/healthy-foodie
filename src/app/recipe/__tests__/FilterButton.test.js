import React from 'react';
import renderer from 'react-test-renderer';
import FilterButton from '../FilterButton';
import fields from '../fieldByFilterButtons';

describe('src/app/recipe:details', () => {
  it('render buttons when filters provided', () => {
    const component = renderer.create(
      <FilterButton items={fields} onClick={e => e.target.value} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('redirect to home when no filters provided', () => {
    const component = renderer.create(
      <FilterButton onClick={e => e.target.value} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
