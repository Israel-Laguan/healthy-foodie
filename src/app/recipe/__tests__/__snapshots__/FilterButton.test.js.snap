// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/recipe:details redirect to home when no filters provided 1`] = `
<div
  className="row"
>
  <div
    className="my-2 col"
  >
    <div
      className="w-100 btn-group"
      role="group"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon undefined"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`src/app/recipe:details render buttons when filters provided 1`] = `
<div
  className="row"
>
  <div
    className="my-2 col"
  >
    <div
      className="w-100 btn-group"
      role="group"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon Vegan"
          src="https://img.icons8.com/doodle/48/000000/no-meat.png"
        />
        Vegan
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon Italian"
          src="https://img.icons8.com/doodle/48/000000/italy.png"
        />
        Italian
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon Chinese"
          src="https://img.icons8.com/doodle/48/000000/china.png"
        />
        Chinese
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon French"
          src="https://img.icons8.com/doodle/48/000000/france.png"
        />
        French
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon Main Course"
          src="https://img.icons8.com/doodle/48/000000/tableware.png"
        />
        Main Course
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon Dessert"
          src="https://img.icons8.com/doodle/48/000000/cupcake--v1.png"
        />
        Dessert
      </button>
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon Salad"
          src="https://img.icons8.com/doodle/48/000000/lettuce--v1.png"
        />
        Salad
      </button>
      <button
        aria-label={null}
        className="btn btn-primary active"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Icon All"
          src="https://img.icons8.com/doodle/48/000000/dining-room.png"
        />
        All
      </button>
    </div>
  </div>
</div>
`;
